.github/workflows/main.yml

```diff
@@ -45,3 +45,39 @@ jobs:
         run: ./llama2-tasks-test
       - name: grok1-tasks-test
         run: ./grok1-tasks-test
+
+  build-windows:
+    name: Windows Build
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+
+      - name: Set up Dependencies
+        run: |
+          choco install make
+
+      - name: Build
+        run: |
+          make dllama
+          make dllama-api
+          make funcs-test
+          make quants-test
+          make transformer-test
+          make llama2-tasks-test
+          make grok1-tasks-test
+
+      - name: Run funcs-test
+        run: .\funcs-test.exe
+
+      - name: Run quants-test
+        run: .\quants-test.exe
+
+      - name: Run transformer-test
```

Comment:
Maybe we don't need to duplicate this section. Please check this post: https://github.com/actions/starter-workflows/issues/154#issuecomment-536721000

Reply:
Also saw I still had an issue with the main.yml, that I've just now fixed.